fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -602,10 +585,16 @@ protected void replaceResourceWithStream(final FedoraResource resource,
     }
 
     protected void patchResourcewithSparql(final FedoraResource resource,
-                                           final String requestBody,
-                                           final RdfStream resourceTriples)
-            throws MalformedRdfException, AccessDeniedException {
-        resource.updateProperties(translator(), requestBody, resourceTriples);
+            final String requestBody,
+            final RdfStream resourceTriples)
+                    throws MalformedRdfException, AccessDeniedException {
+        if (resource instanceof NonRdfSourceDescription) {
+            // update the description instead
```

Comment:
If we are putting all of the description on the binary, then what is the use of the NonRdfSourceDescription? just a URI to GET/PATCH on that under the covers actually interacts with the binary resource properties?
Also, this seems to be in conflict with the following ticket, which is moving in the direction of making the NonRdfSourceDescription a regular container: https://jira.duraspace.org/browse/FCREPO-1590